pulsar-client/src/main/java/org/apache/pulsar/client/api/ConsumerConfiguration.java

```diff
@@ -270,6 +272,27 @@ public void setPriorityLevel(int priorityLevel) {
         this.priorityLevel = priorityLevel;
     }
 
+    public boolean getReadCompacted() {
+        return readCompacted;
+    }
+
+    /**
+     * If enabled, the consumer will read messages from the compacted topic rather than reading the full message
+     * backlog of the topic. This means that, if the topic has been compacted, the consumer will only see the latest
+     * value for each key in the topic, up until the point in the topic message backlog that has been compacted.
+     * Beyond that point, the messages will be sent as normal.
+     *
+     * readCompacted can only be enabled subscriptions to persistent topics, which have a single active consumer
+     * (i.e. failure or exclusive subscriptions). Attempting to enable it on subscriptions to a non-persistent
+     * topics or on a shared subscription, will lead to the subscription call throwing a PulsarClientException.
+     *
+     * @param readCompacted whether to read from the compacted topic
+     */
+    public ConsumerConfiguration setReadCompacted(boolean readCompacted) {
```

Comment:
Can we expect more options to be needed at some point that we could need an enum rather than a flag?

Reply:
I doubt there'd be any more client flags around read compacted. Even if there were, I don't see that an enum would be better here.